### DISCUSSION TITLE
Which theia version maps to which plugin version?

### DISCUSSION THREAD

**Discussion by tstibbs** [eclipse-theia/theia#8284] [General] [answered]
As I write this, the latest theia modules seem to be at version 1.4 (see https://www.npmjs.com/package/@theia/core), but the latest version of the "built-in" markdown language plugin seems to be 1.47.3 (see https://open-vsx.org/extension/vscode/markdown). The version of the plugins seems to map to the VS Code version (which is helpful) but the theia version is a totally different string (which it not so helpful!). How do I know which plugin versions should be used with which theia version?

I am asking because I've got some issues with some plugins, and my first step was to check I'd got compatible versions of those plugins, but this is impossible as I can't find any guidance on how the versions map to each other.

Can anyone help explain?

**Comment by vince-fugnitto** [comment, marked as answer]
@tstibbs there is no direct mapping, for the builtin extensions we identify which APIs are supported and include them in the example applications of this repository. If an extension has problems we downgrade them but there is no direct correlation between Theia extension versions and VS Code plugins (builtins extensions).

**Reply by tstibbs** [reply]
@vince-fugnitto by example applications do you mean https://github.com/theia-ide/theia-apps? Looking at https://github.com/theia-ide/theia-apps/blob/master/theia-full-docker/latest.package.json for example I can see that the version of the node modules is `latest` rather than a specific version, and the version of the plugins is pretty old (`1.39.1` compared with `1.47.3` in openvsx), so I'm not sure what I can infer from that about version compatibility?

**Reply by vince-fugnitto** [reply]
@tstibbs I'm referring to the examples used in this repository for testing purposes when developing the framework (namely [example-browser](https://github.com/eclipse-theia/theia/tree/master/examples/browser) and [example-electron](https://github.com/eclipse-theia/theia/tree/master/examples/electron]). They both use the plugins listed in the [package.json](https://github.com/eclipse-theia/theia/blob/master/package.json).

https://github.com/eclipse-theia/theia/blob/8ac1e5f8d9123196755fa02cb09d7b21eb311457/package.json#L83

We try to use the latest in this repository unless there are compatibility issues as we'd like to ultimately test and work towards API compatibility (but there are of course APIs which have yet to be completed). As I mentioned there is no direct mapping between Theia extensions and VS Code plugin versions, but you may take a look at our 'supported' default API version to get a better understanding of which plugins work and which do not. At the time of writing the default API version is:

https://github.com/eclipse-theia/theia/blob/8ac1e5f8d9123196755fa02cb09d7b21eb311457/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L22

**Reply by tstibbs** [reply]
@vince-fugnitto Most of the plugins in that package.json are at 1.44.2, which according to openvsx is about three months old. I realise that you're saying that you don't test with anything more recent than that, but is the reason for that that anything newer than 1.44.2 requires APIs that theia doesn't provide yet, or realistically is it likely that newer plugins will work but just haven't been tested yet?

**Reply by vince-fugnitto** [reply]
The majority of extensions should work properly, and when support is needed for newer APIs issues are opened to track their compatibility (I don’t think it’s feasible to test every extension). There are some known issues (present in the repo) which track some API improvements and its another reason why extensions are not updated (the extensions in this repo are for test purposes).

The builtins (bundled with VS Code) should work for the majority, and if there are some API incompatibilities it’d be a partial loss of functionality. Your question was about the link between Theia extensions and VS Code plugins and at the moment there is no strict connection.

**Reply by akosyakov** [reply]
We should have a process to bump up `VSCODE_DEFAULT_API_VERSION` regularly and consume latest built-in extensions. Added it to discuss for the dev meeting today.

**Reply by tstibbs** [reply]
Excellent, thank you very much for the detailed responses @vince-fugnitto
